test/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -151,11 +151,24 @@ TEST_P(DynamicForwardProxyIntegrationTest, BasicFlow) {
 
   // There is no buffering in this test, so the first message was dropped. Send another message
   // to verify that it's able to go through after the DNS resolution completed.
-  client.write("hello2", *listener_address);
-
-  Network::UdpRecvData request_datagram;
-  ASSERT_TRUE(fake_upstreams_[0]->waitForUdpDatagram(request_datagram));
-  EXPECT_EQ("hello2", request_datagram.buffer_->toString());
+  // Since the DNS resolution is performed on the main thread, there's a chance that next UDP
+  // datagrams will also be dropped, until the DNS resolution is completed. Therefore, the attempt
+  // to send a datagram will be retried three times.
```

Comment:
I'm OK with this as a fix but is there another stat we could wait on instead?

Reply:
I don't see what stat we could wait on. The test case is waiting for a stat that the DNS entry is loaded, but there's a callback that needs to be called to mark that the DFP filter can unblock next datagrams, here:
https://github.com/envoyproxy/envoy/blob/21bd0df3bced4c95a8318c38aeaf2245ed13cf4e/source/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter.cc#L122

Before this part is called, the next UDP datagrams will still drop. It may be possible to add additional stat just to overcome this, but I'm not sure we should

Reply:
does the DFP filter unblock datagrams because there's change to clusters or hosts?  If so there's a lot of stats there which might be useful.  I often apply a local patch below to see if there are any existing stats which are relevant. If you run this and confirm there's none this can land as-is.

--- a/source/common/stats/allocator_impl.cc
+++ b/source/common/stats/allocator_impl.cc
@@ -150,6 +150,7 @@ public:
 
   // Stats::Counter
   void add(uint64_t amount) override {
+     std::cerr << "Adding to " << name() << "\n";
     // Note that a reader may see a new value but an old pending_increment_ or
     // used(). From a system perspective this should be eventually consistent.
     value_ += amount;

Reply:
Tried printing, but don't see another stat that is relevant and could resolve the flakiness